NeoMathEngine/src/CMakeLists.txt

```diff
@@ -91,6 +91,7 @@ target_sources(${PROJECT_NAME}
     ../include/NeoMathEngine/LookupData.h
     ../include/NeoMathEngine/MemoryHandle.h
     ../include/NeoMathEngine/MemoryHandle.inl
+    ../include/NeoMathEngine/MathEngineDropout.h
```

Comment:
It is not very good to public this header to use interface.
Maybe you should refuse to check `isValid` in NeoML code and hide it somehow into MathEngine code or just not to check at all.
I sure, that `MathEngineDropout.h` should be moved to `src` folder.

Reply:
moved everything to src files, so only CDropoutDesc accessible outside mathengine
left desc fields public for mathengine dropout function (no access from outside anyway)

Reply:
Super!